x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/ElasticsearchMappings.java

```diff
@@ -100,7 +101,7 @@ public class ElasticsearchMappings {
 
     private ElasticsearchMappings() {}
 
-    static String[] mappingRequiresUpdate(ClusterState state, String[] concreteIndices, Version minVersion) {
+    static String[] mappingRequiresUpdate(ClusterState state, String[] concreteIndices, MlConfigVersion minVersion) {
```

Comment:
This function is used with ML system indices as well as hidden indices. System index versioning needs to be understood by core server code. So I don't think we can use `MlConfigVersion` for versioning our index mappings. We'll have to use the approach of https://github.com/elastic/elasticsearch/pull/97529.

There's more detail in https://github.com/elastic/elasticsearch/pull/97529#discussion_r1268181542 and my reply to that.

I think it's best if you change this PR to leave index mapping versioning as it was, and just change our config and state versioning. Sorry for the extra work.